pkg/state/appender.go

```diff
@@ -790,6 +791,87 @@ func (a *txAppender) appendFixSnapshots(fixSnapshots []proto.AtomicSnapshot, blo
 	return nil
 }
 
+func (a *txAppender) isLastBlockFinalized(
+	finalizationVoting *proto.FinalizationVoting,
+	block *proto.BlockHeader,
+	height proto.Height,
+) (bool, error) {
+
+	if finalizationVoting == nil {
+		return false, nil
+	}
+	// TODO what if it's empty?
+	lastFinalized, err := a.stor.finalizations.newest()
+	if err != nil {
+		return false, err
+	}
+	msg, err := proto.EndorsementMessage(
```

Comment:
Why construct an endorsement message here?

Reply:
Ah, I see, it's for block header validation. I'd rather move this function to proto package, somewhere near block declaration or even made it a part of Block type.

Reply:
Because this function `EndorsementMessage` is already in proto